apps/lib/opt.c

```diff
@@ -610,38 +599,10 @@ int opt_uintmax(const char *value, ossl_uintmax_t *result)
         errno = oerrno;
         return 0;
     }
-    /* Ensure that the value in |m| is never too big for |*result| */
-    if (sizeof(m) > sizeof(*result)
-        && m > OSSL_UINTMAX_MAX) {
-        opt_number_error(value);
-        return 0;
-    }
-    *result = (ossl_intmax_t)m;
+    *result = (intmax_t)m;
```

Comment:
```suggestion
    *result = m;
```
oh, i missed that the cast was also wrong previously :)

Reply:
this, too.